### PR TITLE
Get providers for tx hashes from Firebase

### LINKED ISSUE #100: Read providers of tx hashes from Firebase
### Expected Behavior

We're going to be assigning providers to tx hashes according to the info we get from webhooks from our providers. That info will be uploaded to Firebase under `registrations/{address}/txHashes`. The app should read from there and show the correct provider name and picture on the feed.

### Current Behavior

The app doesn't read from Firebase info about any tx hash.

### PR DESCRIPTION
**Pull request by gnardini**
### Description

On Firebase, we now read from a mapping on `registrations/{address}/txHashes` from `txHash` to `provider`, which is one of `MOONPAY`, `SIMPLEX`, `RAMP` or `TRANSAK` for now. We use that information to show the correct name and picture on the feed.

### Other changes

- Refactored the `SELECT_PROVIDER` action a bit to use an enum instead of name & picture.

### Tested

Manually and added unit test.

### Related issues

- Fixes #100

### Backwards compatibility

Previous versions won't see the providers on their feed but nothing should break

### CODE REVIEW

File: packages/mobile/src/fiatExchanges/reducer.ts [hunk 1]
```diff
@@ -5,14 +5,52 @@ import i18n from 'src/i18n'
 import { getRehydratePayload, REHYDRATE } from 'src/redux/persist-helper'
 import { RootState } from 'src/redux/reducers'
 
+export enum CiCoProvider {
+  MOONPAY = 'MOONPAY',
+  RAMP = 'RAMP',
+  SIMPLEX = 'SIMPLEX',
+  TRANSAK = 'TRANSAK',
+}
```
Comment (jeanregisser) [inline packages/mobile/src/fiatExchanges/reducer.ts:13]: nit: we have a proposed naming for enum: https://celo-org.slack.com/archives/CL6JFBALE/p1589448552015000

File: packages/mobile/src/fiatExchanges/reducer.ts [hunk 2]
```diff
@@ -31,19 +69,22 @@ export const reducer = (state: State = initialState, action: ActionTypes | Rehyd
...
-        },
+        lastUsedProvider: action.provider,
       }
     case Actions.ASSIGN_PROVIDER_TO_TX_HASH:
-      let providerToAssign = state.lastUsedProvider
-      if (!providerToAssign) {
+      // Don't override in case it came from Firebase (Actions.SET_PROVIDERS_FOR_TX_HASHES).
+      if (state.txHashToProvider[action.txHash]) {
+        return state
+      }
```
Comment (jeanregisser) [inline packages/mobile/src/fiatExchanges/reducer.ts:78]: I don't understand the comment. If action is `Actions.SET_PROVIDERS_FOR_TX_HASHES` why would it be handled here?

File: packages/mobile/src/fiatExchanges/reducer.ts [hunk 3]
```diff
@@ -5,14 +5,52 @@ import i18n from 'src/i18n'
...
 }
 
+interface TxHashToDisplayInfo {
+  [txHash: string]: ProviderFeedInfo | undefined
+}
+
 export interface State {
-  lastUsedProvider: ProviderFeedInfo | null
-  txHashToProvider: { [txHash: string]: ProviderFeedInfo | undefined }
+  lastUsedProvider: CiCoProvider | null
```
Comment (jeanregisser) [inline packages/mobile/src/fiatExchanges/reducer.ts:52]: I haven't seen a migration for this. Do we need one?

**Review by jeanregisser — COMMENTED** [review]
Great!

On a minor note, this would link addresses to CICO providers operations they made, in a public way since anyone can read it with the current rules.
Probably fine for now, but would be nice for this to stay private in the future.

cc @nambrot @nityas @aleksey-nov

**Comment by gnardini** [comment]
> On a minor note, this would link addresses to CICO providers operations they made, in a public way since anyone can read it with the current rules.

This is being stored on `registrations/${address}`, which should only be readable by the address owner.

File: packages/mobile/src/fiatExchanges/reducer.ts [hunk 4]
```diff
@@ -5,14 +5,52 @@ import i18n from 'src/i18n'
...
+  },
+  [CiCoProvider.SIMPLEX]: {
+    name: 'Simplex',
+    icon:
+      'https://firebasestorage.googleapis.com/v0/b/celo-mobile-mainnet.appspot.com/o/images%2Fsimplex.jpg?alt=media&token=6037b2f9-9d76-4076-b29e-b7e0de0b3f34',
+  },
+  [CiCoProvider.TRANSAK]: {
+    name: 'Transak',
+    icon:
+      'https://storage.cloud.google.com/celo-mobile-mainnet.appspot.com/images/transak-icon.png',
```
[the remark this answers is not in the document]
Comment (gnardini) [reply packages/mobile/src/fiatExchanges/reducer.ts:34]: Not sure, switched to firebase storage

File: packages/mobile/src/fiatExchanges/reducer.ts (hunk 2 above)
Comment (gnardini) [reply packages/mobile/src/fiatExchanges/reducer.ts:78]: I can see how the comment was confusing 😅 

We have two ways of assigning providers: The 'good way' that eventually will be the only way which is reading from Firebase (which is handled with the `SET_PROVIDERS_FOR_TX_HASHES` action) or assigning a tx hash that comes from a known partner address (the Bitrex ones) to the last provider clicked, which is of course much less accurate but it was the only way we had until now (and is still the only way for Simplex).

What I meant with the comment is that if a tx hash was already assigned using the 'good way' (through Firebase) we wouldn't want to override it with the error-prone temporary way.

I changed the comment to something more generic, let me know if you think I should be more specific with the comment.

File: packages/mobile/src/fiatExchanges/reducer.ts (hunk 3 above)
Comment (gnardini) [reply packages/mobile/src/fiatExchanges/reducer.ts:52]: Yeah, good point. I think it's fine to lose the info, but we should make sure it's set to null and doesn't have info formatted in an unexpected way.

**Comment by jeanregisser** [comment]
What's NOT backward compatible with this change?

**Comment by gnardini** [comment]
@jeanregisser Sorry, not sure why I put no there, all I meant is that previous versions won't see the providers on their feed, which is expected :)

**Review by jeanregisser — APPROVED** [review]

**Review by tarikbellamine — APPROVED** [review]
LGTM!

**Pull request merged**